### PR TITLE
feat: clearing role during user deletion

### PR DESCRIPTION
**Pull request by voikin**
BUG `ASP-1268`

### CODE REVIEW

**Comment by dplynskiy** [comment]
I think we should make this check in DeleteUser in object package, not in controllers

File: controllers/user.go [hunk 1]
```diff
@@ -736,6 +736,36 @@ func (c *ApiController) DeleteUser() {
 		return
 	}
 
+	err = object.ExtendUserWithRolesAndPermissions(userFromDb)
```
Comment (svgrafov) [inline controllers/user.go:739]: Maybe move this addition from the controller level, a bit deeper?

File: controllers/util.go [hunk 2]
```diff
@@ -334,3 +334,12 @@ func getInvalidSmsReceivers(smsForm SmsForm) []string {
 	}
 	return invalidReceivers
 }
+
+func removeElementFromSlice[T comparable](slice []T, element T) []T {
```
Comment (svgrafov) [inline controllers/util.go:338]: Please add tests for this.

Comment (dplynskiy) [reply controllers/util.go:338]: we already have DeleteVal func in util package, mb use it instead new func?

**Review by svgrafov — APPROVED** [review]

**Pull request merged**